### PR TITLE
Add an option to ignore (block) a user when reporting their events

### PR DESCRIPTION
**Pull request by turt2live**
This is primarily useful if the content being reported really doesn't belong on your screen, and the room moderators are slow to react.

Ideally we'd use the word "block" instead of "ignore", but we call it "ignore user" everywhere else. See https://github.com/vector-im/element-web/issues/19590 for further context on the word choice.

This change includes a minor refactor to the styles of labelled toggles (for reusability).

image

---
Here's what your changelog entry will look like:

## ✨ Features
 * Add an option to ignore (block) a user when reporting their events ([\#8471](https://github.com/matrix-org/matrix-react-sdk/pull/8471)).

### CODE REVIEW

File: src/components/views/elements/LabelledToggleSwitch.tsx [hunk 1]
```diff
@@ -52,7 +53,10 @@ export default class LabelledToggleSwitch extends React.PureComponent<IProps> {
             secondPart = temp;
         }
 
-        const classes = `mx_SettingsFlag ${this.props.className || ""}`;
+        const classes = classNames("mx_SettingsFlag", {
+            [this.props.className ?? ""]: true,
```
Comment (t3chguy) [inline src/components/views/elements/LabelledToggleSwitch.tsx:57]:
```suggestion
        const classes = classNames("mx_SettingsFlag", this.props.className, {
```

File: src/components/views/dialogs/ReportEventDialog.tsx [hunk 2]
```diff
@@ -42,6 +43,7 @@ interface IState {
     err?: string;
     // If we know it, the nature of the abuse, as specified by MSC3215.
     nature?: ExtendedNature;
+    ignoreUserToo: boolean; // if true, user will be ignored/blocked on submit
```
Comment (t3chguy) [inline src/components/views/dialogs/ReportEventDialog.tsx:46]: not sure how I feel about the `too` - `ignoreUser` should be fine, no?

**Review by t3chguy — APPROVED** [review]
LGTM otherwise

File: src/components/views/dialogs/ReportEventDialog.tsx (hunk 2 above)
Comment (turt2live) [reply src/components/views/dialogs/ReportEventDialog.tsx:46]: it just didn't feel right to not acknowledge what the flag was doing. Given the context of the dialog, "ignore user" could mean "instead", "in addition to", or "don't file report". If we used the word "block" throughout the application instead, it'd be more than fine to leave it at `blockUser`.

**Review by janogarcia — CHANGES_REQUESTED** [review]
@turt2live Toggles, in general, should be used for actions that take place immediately upon acting on them.

Thus, for this case (enabled on form submission), I'd suggest changing it to a checkbox with supporting text as in the mockup below.

Frame 6501

**Comment by turt2live** [comment]
sounds good @janogarcia - should we also add a backlog item to use the same design for areas we already use a toggle switch? (mostly widget permissions stuff)

**Comment by janogarcia** [comment]
This will eventually be addressed as part of the the design systems improvements I'm working on, along with other changes. Already on my list but feel free to file an issue for it.

File: src/components/views/elements/LabelledToggleSwitch.tsx (hunk 1 above)
Comment (turt2live) [reply src/components/views/elements/LabelledToggleSwitch.tsx:57]: applied manually.

**Comment by turt2live** [comment]
@janogarcia new look:

image

let me know if further changes need to be made.

**Review by t3chguy — APPROVED** [review]

**Comment by janogarcia** [comment]
@turt2live There's no Netlify build available this time so, based on your screenshot:

- Use `$primary-content` color for the checkbox label (`Ignore user`)
- Use `$secondary-content` color for the checkbox supporting text and the modal body text (`Reporting this message will send its…`).

There's a lot more stuff that we could improve in this modal (copy, visual design), and modals in general, but I'm trying to keep the scope reasonable, as upcoming redesigns will address that.

**Review by janogarcia — CHANGES_REQUESTED** [review]
So, there's a 99.9% probability that I'll forget to send my PR comments via the dedicated form:

Please refer to my previous comment
https://github.com/matrix-org/matrix-react-sdk/pull/8471#issuecomment-1129758494

**Comment by t3chguy** [comment]
@janogarcia 
image

I can see the Netlify build from the last change 6 days ago

image

**Comment by janogarcia** [comment]
@t3chguy Ouch, thanks! I always check the `Merge state` section on the bottom and look for the Netlify link in the PR description. Could it be that the bot that adds the link to the description stopped working?

**Comment by janogarcia** [comment]
@t3chguy That said, I can't seem to find the [deployment status on your second screenshot](https://user-images.githubusercontent.com/2403652/169002477-01c5b738-2654-4ae7-b96f-1f9ea7750e98.png) anywhere here.

**Comment by t3chguy** [comment]
@janogarcia do you not see this section at the bottom?

image

**Comment by janogarcia** [comment]
The deployment section seems to be missing on my end:

Add an option to ignore (block) a user when reporting their events by turt2live · Pull Request #8471 · matrix-org-matrix-react-sdk

**Comment by t3chguy** [comment]
@janogarcia strange, can't find any github documentation on whether this section needs permission to show, the `View deployment` in the PR timeline should be accessible though, I can see it even in incognito

**Comment by janogarcia** [comment]
@turt2live Makes sense. Let's keep it consistently bad for now (better to have one rather than two suboptimal implementations 🙃). But I promise to come back to this at some point and end with all the awful debt once and for all. 🪄✨👹

**Comment by janogarcia** [comment]
> the View deployment in the PR timeline should be accessible though, I can see it even in incognito

@t3chguy Yeah, that even remains visible.

**Review by janogarcia — APPROVED** [review]
@turt2live I re-requested a review for myself. Is that how it's supposed to work?

**Comment by turt2live** [comment]
Should be able to go to the Files Changed tab -> Review Changes dropdown -> Approve radio button -> Submit

(doesn't necessarily require a review request, but good to have one)

**Comment by turt2live** [comment]
oh, you did that. Ignore me - thanks!

**Pull request merged**